platforms/ide/tooling-api/src/crossVersionTest/groovy/org/gradle/integtests/tooling/r87/ProblemProgressEventCrossVersionTest.groovy

```diff
@@ -0,0 +1,279 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.integtests.tooling.r87
+
+
+import org.gradle.integtests.fixtures.GroovyBuildScriptLanguage
+import org.gradle.integtests.tooling.fixture.TargetGradleVersion
+import org.gradle.integtests.tooling.fixture.ToolingApiSpecification
+import org.gradle.integtests.tooling.fixture.ToolingApiVersion
+import org.gradle.tooling.BuildException
+import org.gradle.tooling.events.problems.FileLocation
+import org.gradle.tooling.events.problems.LineInFileLocation
+import org.gradle.tooling.events.problems.OffsetInFileLocation
+import org.gradle.tooling.events.problems.ProblemAggregationDescriptor
+import org.gradle.tooling.events.problems.ProblemDescriptor
+import org.gradle.tooling.events.problems.Severity
+import org.gradle.tooling.events.problems.TaskPathLocation
+import org.gradle.tooling.events.problems.internal.DefaultProblemsOperationDescriptor
+
+import static org.gradle.integtests.tooling.r86.ProblemProgressEventCrossVersionTest.getProblemReportTaskString
+
+@ToolingApiVersion(">=8.7")
+@TargetGradleVersion(">=8.7")
+class ProblemProgressEventCrossVersionTest extends ToolingApiSpecification {
+
+    def withReportProblemTask(@GroovyBuildScriptLanguage String taskActionMethodBody) {
+        buildFile getProblemReportTaskString(taskActionMethodBody)
+        // TODO using the following code breaks the test, but it should be possible to use it
+        //  buildFile getProblemReportingScript(taskActionMethodBody)
+        //  issue https://github.com/gradle/gradle/issues/27484
+    }
+
+    @TargetGradleVersion("=8.5")
+    def "test failure context"() {
+        setup:
+        buildFile << """
+            plugins {
+              id 'java-library'
+            }
+            repositories.jcenter()
+            task bar {}
+            task baz {}
+        """
+
+
+        when:
+        def listener = new org.gradle.integtests.tooling.r85.ProblemsServiceModelBuilderCrossVersionTest.ProblemProgressListener()
+        withConnection { connection ->
+            connection.newBuild()
+                .forTasks(":ba")
+                .addProgressListener(listener)
+                .setStandardError(System.err)
+                .setStandardOutput(System.out)
+                .addArguments("--info")
+                .run()
+        }
+
+        then:
+        thrown(BuildException)
+        listener.problems.size() == 2
+    }
+
+    def "Problems expose details via Tooling API events"() {
```

Comment:
This and the tests below already worked with TAPI 8.6 -> Gradle 8.6 and so they belong to the `r86` package, without the failure assertion of course. The tests should be moved back to the `r86` package and then have a simplified variant here, asserting the exception field.

Reply:
The newer one also has to check for `failure` to my understanding.